Homework1

ne-kit-28 opened the pull request.

ne-kit-28 committed:
homework1 has been planted

ne-kit-28 committed:
README was updated

ne-kit-28 committed:
Update README.md

ne-kit-28 committed:
Update README.md

YlabCourse commented on homework1/.idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Папки .idea в репозитории быть не должно, убери ее.

YlabCourse commented on homework1/src/main/java/y_lab/Main.java:
```diff
@@ -0,0 +1,60 @@
+        ProgressReportUseCase progressReportUseCase = new ProgressReportUseCase(generateProgressStatisticsUseCase, streakCalculationUseCase);
+        RegistrationUseCase registrationUseCase = new RegistrationUseCase(userRepository);
+        UpdateHabitUseCase updateHabitUseCase = new UpdateHabitUseCase(habitRepository);
+        GetUsersUseCase getUsersUseCase = new GetUsersUseCase(userRepository);
```
Все классы инициализируются в main. Чтобы этого избежать предлагаю использовать паттерн фабрика для создания экземпляров своих классов.

YlabCourse commented on homework1/src/main/java/y_lab/Main.java:
```diff
@@ -0,0 +1,60 @@
+        executorService.startScheduler();
+        inputAdapter.start();
+        userRepository.saveToFile("Users.ser");
+        //userRepository.saveAdmins("Admins.ser");
```
Таких закомментированных кусков кода не должно быть в ПР

YlabCourse commented on homework1/src/main/java/y_lab/domain/repositories/HabitRepository.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.ArrayList;
+import java.util.Optional;
+
+public interface HabitRepository {
```
javadoc принято размещать в интерфейсах, а в реализациях можно не указывать.

YlabCourse commented on homework1/src/main/java/y_lab/domain/repositories/HabitRepository.java:
```diff
@@ -0,0 +1,16 @@
+package y_lab.domain.repositories;
```
Названия пакетов - в единственном числе

YlabCourse commented on homework1/src/main/java/y_lab/out/repositories/HabitRepositoryImpl.java:
```diff
@@ -0,0 +1,167 @@
+
+    /**
+     * Stores habit entities using their ID as the key.
+     */
```
Перенеси javadoc в интерфейс, это не будет отвлекать при чтении кода реализации.

YlabCourse commented on homework1/src/main/java/y_lab/out/repositories/HabitRepositoryImpl.java:
```diff
@@ -0,0 +1,167 @@
+ */
+@Setter
+@Getter
+public class HabitRepositoryImpl implements HabitRepository, DataService {
```
Нарушен принцип единой ответственности. Репозиторий должен хранить сущности, за запись в файл пусть отвечает отдельный класс. Раздели логику хранения/получения из памяти и записи/чтения файла. Сама идея репозитория, реализующего одновременно репозиторий и сервис как-то порочно выглядит.

YlabCourse commented on homework1/src/main/java/y_lab/out/repositories/HabitRepositoryImpl.java:
```diff
@@ -0,0 +1,167 @@
+package y_lab.out.repositories;
```
В пакете out оставь класс, отвечающий за запись в файл. Реализацию репозитория унеси в пакет impl внутри пакета repository (создай его)

YlabCourse commented on homework1/src/main/java/y_lab/out/repositories/ProgressRepositoryImpl.java:
```diff
@@ -0,0 +1,146 @@
+ */
+@Getter
+@Setter
+public class ProgressRepositoryImpl implements ProgressRepository, DataService {
```
Здесь и далее - аналогично. Раздели логику, распредели классы по пакетам.

YlabCourse commented on homework1/src/main/java/y_lab/service/EmailValidatorService.java:
```diff
@@ -0,0 +1,30 @@
+ * Service for validating email addresses.
+ * This class provides a method to check the validity of email formats based on a regex pattern.
+ */
+public class EmailValidatorService {
```
Больше тянет на утилитный класс-валидатор, чем на целый сервис.

YlabCourse commented on homework1/src/main/java/y_lab/service/ExecutorService.java:
```diff
@@ -0,0 +1,65 @@
+ * Service for scheduling notifications for habits.
+ * This class manages the scheduling of notifications based on the frequency of habits.
+ */
+public class ExecutorService {
```
Сервисы лучше все же через интерфейсы реализовать. И javadoc там оставить. Создай вложенный пакет impl и все реализации помести туда.

YlabCourse commented on homework1/src/main/java/y_lab/in/Adapters/ConsoleAdministratorInputAdapter.java:
```diff
@@ -0,0 +1,170 @@
+        Long otherUserId;
+
+        while (true) {
+            System.out.println("Select option:");
+            System.out.println("1. Get users");
+            System.out.println("2. Edit profile");
+            System.out.println("3. Delete account");
+            System.out.println("4. Log out");
```
Здесь и в подобных случаях можно использовать текстовые блоки, чтобы на вывод отдавать один раз, а не построчно. Сами текстовые блоки лучше вынести в отдельный класс констант.

YlabCourse commented on homework1/src/test/java/y_lab/usecases/CreateHabitUseCaseTest.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    @Test
+    void createHabit_shouldCreateHabit_whenHabitDoesNotExist() {
```
Используй @DisplayName для описания методов и классов в тестах

ne-kit-28 committed:
The comments have been eliminated

ne-kit-28 committed:
B
Merge branch 'homework1' of https://github.com/ne-kit-28/y_lab_homework into homework1

ne-kit-28 committed:
gitIgnore edite

ne-kit-28 committed:
gitIgnore edite

ne-kit-28 committed:
.idea was deleted

ne-kit-28 committed:
Merge branch 'main' into homework1
